common/components/WalletDecrypt/ViewOnly.tsx

```diff
@@ -1,29 +1,60 @@
 import React, { Component } from 'react';
 import translate from 'translations';
+import { donationAddressMap } from 'config/data';
+import { isValidETHAddress } from 'libs/validators';
+import { AddressOnlyWallet } from 'libs/wallet';
+
+interface Props {
+  onUnlock(param: any): void;
+}
+
+interface State {
+  address: string;
+}
+
+export default class ViewOnlyDecrypt extends Component<Props, State> {
+  public state = {
+    address: ''
+  };
 
-export default class ViewOnlyDecrypt extends Component {
   public render() {
+    const { address } = this.state;
+
     return (
       <section className="col-md-4 col-sm-6">
         <div id="selectedUploadKey">
-          <h4>
-            {translate('ADD_Radio_2_alt')}
-          </h4>
-
-          <div className="form-group">
-            <input type="file" id="fselector" />
-
-            <a
-              className="btn-file marg-v-sm"
-              id="aria1"
-              tabIndex={0}
-              role="button"
-            >
-              {translate('ADD_Radio_2_short')}
-            </a>
-          </div>
+          <h4>{translate('MYWAL_Address')}</h4>
+
+          <form className="form-group" onSubmit={this.openWallet}>
+            <input
+              className={`form-control
+                ${isValidETHAddress(address) ? 'is-valid' : 'is-invalid'}
+              `}
+              onChange={this.changeAddress}
+              value={address}
+              placeholder={donationAddressMap.ETH}
+            />
+
+            <button className="btn btn-primary btn-block">
+              {translate('NAV_ViewWallet')}
+            </button>
+          </form>
         </div>
       </section>
     );
   }
+
+  private changeAddress = (ev: React.FormEvent<HTMLInputElement>) => {
+    this.setState({ address: ev.currentTarget.value });
+  };
+
+  private openWallet = (ev: React.SyntheticEvent<HTMLFormElement>) => {
+    const { address } = this.state;
+    ev.preventDefault();
+    if (isValidETHAddress(address)) {
```

Comment:
Would it be better here to disable the button outright if its not a valid address? Then you dont need to check validity before calling the unlock prop

Reply:
I'll leave the check here, since it's pretty inexpensive, but I'll definitely add disabling the button.

Reply:
This is done via 1d7c7b89a2affd0653c092ed6bfd3edd574cba2f